olive/evaluator/olive_evaluator.py

```diff
@@ -748,15 +753,20 @@ def _inference(
             # it is expensive to convert to list and then convert back to torch tensor
             preds.append(outputs.cpu())
             targets.append(labels.cpu())
-            logits.append(
-                result.logits.cpu()
-                if not isinstance(result, torch.Tensor) and getattr(result, "logits", None) is not None
-                else result.cpu()
-            )
+            if isinstance(result, torch.Tensor):
+                logits.append(result.cpu())
+            elif isinstance(result, (list, tuple)):
+                logits.append([r.cpu() for r in result])
+            elif isinstance(result, dict):
+                for k in result:
+                    logits_dict[k].append(result[k].cpu())
```

Comment:
How can we handle the case when result has logits attribute?

Reply:
You mean result like `{ "logits": tensor }`?